OpenEdXMobile/src/main/java/org/edx/mobile/view/dialog/CourseModalDialogFragment.kt

```diff
@@ -110,6 +105,33 @@ class CourseModalDialogFragment : DialogFragment() {
         }
     }
 
+    private fun trackEvents() {
+        iapAnalytics.initCourseValues(
+            courseId = courseId,
+            isSelfPaced = isSelfPaced,
+            screenName = screenName
+        )
+        var experimentGroup: String? = null
+        if (environment.appFeaturesPrefs.isIAPExperimentEnabled()) {
+            experimentGroup =
+                if (environment.loginPrefs.isOddUserId) Analytics.Values.TREATMENT else Analytics.Values.CONTROL
+        }
+        environment.analyticsRegistry.trackValuePropLearnMoreTapped(courseId, screenName)
+        environment.analyticsRegistry.trackScreenView(
+            Events.VALUE_PROP_MODAL_VIEW,
+            courseId,
+            null,
+            mapOf(Pair(KEY_SCREEN_NAME, screenName))
+        )
+        environment.analyticsRegistry.trackValuePropMessageViewed(
```

Comment:
as there is a track event with each screen event so we have to use only `trackScreenView`